tools/smithy-rs-tool-common/src/changelog.rs

```diff
@@ -173,16 +176,91 @@ pub struct SdkModelEntry {
 #[derive(Clone, Default, Debug, Deserialize, Serialize)]
 pub struct Changelog {
     #[serde(rename = "smithy-rs")]
-    #[serde(default)]
+    #[serde(default, deserialize_with = "deserialize_smithy_handauthored")]
     pub smithy_rs: Vec<HandAuthoredEntry>,
     #[serde(rename = "aws-sdk-rust")]
-    #[serde(default)]
+    #[serde(default, deserialize_with = "deserialize_rust_sdk_handauthored")]
     pub aws_sdk_rust: Vec<HandAuthoredEntry>,
     #[serde(rename = "aws-sdk-model")]
     #[serde(default)]
     pub sdk_models: Vec<SdkModelEntry>,
 }
 
+fn deserialize_smithy_handauthored<'de, D>(
+    deserializer: D,
+) -> Result<Vec<HandAuthoredEntry>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    deserialize_handauthored(
+        deserializer,
+        |entry: &mut HandAuthoredEntry| -> Result<(), anyhow::Error> {
+            // a change that does not have an affected SDK written against it, shall be
+            // considered a change in ALL sdks
+            if entry.meta.target.is_none() {
+                entry.meta.target = Some(SdkAffected::All);
+            }
+            Ok(())
+        },
+    )
+}
+fn deserialize_rust_sdk_handauthored<'de, D>(
+    deserializer: D,
+) -> Result<Vec<HandAuthoredEntry>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    deserialize_handauthored(
+        deserializer,
+        |entry: &mut HandAuthoredEntry| -> Result<(), anyhow::Error> {
+            if entry.meta.target.is_some() {
+                bail!("cannot have target SDK affected as metadata")
+            }
+            Ok(())
+        },
+    )
+}
+
+fn deserialize_handauthored<'de, D, F>(
```

Comment:
Deserialisation code has been removed and moved into parse_str now. The premise of implementing at the deserialisation stage was 1) to generate errors early in the pipeline 2) an incorrect value in a toml filed should result in an error at the same stage as the error for missing field does. e.g. a hand authored entry with a missing author field results in errors when `toml::from_str` is called. 3) the code should not rely on the caller to call a separate validation function, because if they forget to call it they would be working with incorrect data 4) calling a validation function should be an immutable and repeatable operation where as we need to overwrite missing meta.target values with SdkAffected::All.

The new code calls validation as part of `Changelog::parse_str`, so that point number 2,3 and 4 in the above paragraph can still be guaranteed. However, if someone calls `toml::from_str` to load the Changelog then `parse_str` is not called  and they will have to call validation separately.